tests-integration/extensions_test.go

```diff
@@ -443,6 +444,53 @@ func TestExtensionToDevice(t *testing.T) {
 	m.MatchResponse(t, res, m.MatchList("a", m.MatchV3Count(0)), m.MatchToDeviceMessages(newToDeviceMsgs))
 }
 
+// Test that if you sync with a very very high numbered since value, we return lower numbered entries.
+// This guards against dropped databases.
+func TestExtensionToDeviceSequence(t *testing.T) {
+	pqString := testutils.PrepareDBConnectionString()
+	// setup code
+	v2 := runTestV2Server(t)
+	v3 := runTestServer(t, v2, pqString)
+	defer v2.close()
+	defer v3.close()
+	alice := "@TestExtensionToDeviceSequence_alice:localhost"
+	aliceToken := "ALICE_BEARER_TOKEN_TestExtensionToDeviceSequence"
+	v2.addAccount(t, alice, aliceToken)
+	toDeviceMsgs := []json.RawMessage{
+		json.RawMessage(`{"sender":"alice","type":"something","content":{"foo":"1"}}`),
+		json.RawMessage(`{"sender":"alice","type":"something","content":{"foo":"2"}}`),
+		json.RawMessage(`{"sender":"alice","type":"something","content":{"foo":"3"}}`),
+		json.RawMessage(`{"sender":"alice","type":"something","content":{"foo":"4"}}`),
+	}
+	v2.queueResponse(alice, sync2.SyncResponse{
+		ToDevice: sync2.EventsResponse{
+			Events: toDeviceMsgs,
+		},
+	})
+
+	hiSince := "999999"
+	res := v3.mustDoV3Request(t, aliceToken, sync3.Request{
+		Lists: map[string]sync3.RequestList{"a": {
+			Ranges: sync3.SliceRanges{
+				[2]int64{0, 10}, // doesn't matter
+			},
+		}},
+		Extensions: extensions.Request{
+			ToDevice: &extensions.ToDeviceRequest{
+				Core:  extensions.Core{Enabled: &boolTrue},
+				Since: hiSince,
+			},
+		},
+	})
+	m.MatchResponse(t, res, m.MatchList("a", m.MatchV3Count(0)), m.MatchToDeviceMessages(toDeviceMsgs), func(res *sync3.Response) error {
+		// ensure that we return a lower numbered since token
+		if res.Extensions.ToDevice.NextBatch == hiSince {
+			return fmt.Errorf("next_batch got %v wanted lower", hiSince)
+		}
```

Comment:
I agree with the comment here.
But the check isn't actually checking for a lower number.